src/main/java/hudson/scm/PerJobCredentialStore.java

```diff
@@ -105,8 +112,24 @@ private Object writeReplace() {
     private static final ThreadLocal<Boolean> IS_SAVING = new ThreadLocal<Boolean>();
 
     /*package*/ void migrateCredentials(SubversionSCM.DescriptorImpl descriptor) throws IOException {
-        for (Map.Entry<String, Credential> e : credentials.entrySet()) {
-            descriptor.migrateCredentials(project, e.getKey(), e.getValue());
+        Iterable<CredentialsStore> it = CredentialsProvider.lookupStores(project);
+        if (it != null && it.iterator().hasNext()) {
```

Comment:
guava's `Iterables#getFirst` may be an nice way to improve readability on this one.